docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,200 @@
+
+```
+type Options struct {
+	Port uint32
+}
+```
+
+The `Network` type holds network-specific properties (i.e. configuration values) and adapters. Ready-to-use Cosmos SDK release series-specific base `Network` concrete types will be made available. Applications can embed such types in their own custom types.
+
+```
+type Network struct {
+	Properties rosetta.NetworkProperties
+	Adapter    rosetta.Adapter
+}
+````
+
+A `NetworkProperties` `struct` comprises basic values that are required by a Rosetta API `Service` to run:
+
+```
+type NetworkProperties struct {
+	// Mandatory properties
+	Blockchain          string
+	Network             string
+	AddrPrefix          string
+	SupportedOperations []string
+}
+```
+
+Rosetta API services use Blockchain and Network as identifiers, e.g. the developers of gaia, the application that powers the Cosmos Hub, may want to set those to Cosmos Hub and cosmos-hub-3 respectively.
+
+`AddrPrefix` contains the network-specific address prefix. Cosmos SDK base implementations will default to `cosmos`, client applications are instructed that this should be changed according to their network configuration.
+
+`SupportedOperations` contains the transaction types that are supported by the library. At the present time, only `Transfer` is supported.
+
+`Network` holds an `Adapter` reference too. Adapter implementations may vary across different Cosmos SDK release series:
+
+```
+type Adapter interface {
+	DataAPI
+	ConstructionAPI
+}
+
+type DataAPI interface {
+	server.NetworkAPIServicer
+	server.AccountAPIServicer
+	server.MempoolAPIServicer
+	server.BlockAPIServicer
+	server.ConstructionAPIServicer
+}
+
+type ConstructionAPI interface {
+	server.ConstructionAPIServicer
+}
+
+````
+
+### Cosmos SDK Integration
+
+Cosmos SDK provides a base `Network` struct and a `NewNetwork` constructor that could serve as code example for client application developers and testing tool to be used in conjuction with `simd`.
+
+```
+// NewNetwork returns the default application configuration.
+func NewNetwork(options Options) service.Network {
+	cosmosClient := cosmos.NewClient(fmt.Sprintf("http://%s", options.CosmosEndpoint))
+	tendermintClient := tendermint.NewClient(fmt.Sprintf("http://%s", options.TendermintEndpoint))
+
+	return service.Network{
+		Properties: rosetta.NetworkProperties{
+			Blockchain:          options.Blockchain,
+			Network:             options.Network,
+			SupportedOperations: []string{OperationTransfer},
+		},
+		Adapter: newAdapter(
+			cosmosClient,
+			tendermintClient,
+			properties{
+				Blockchain:  options.Blockchain,
+				Network:     options.Network,
+				AddrPrefix:  options.AddrPrefix,
+				OfflineMode: options.OfflineMode,
+			},
+		),
+	}
+}
+```
+
+#### In-process Execution
+
+Rosetta API service could run within the same execution process of the application; new configuration option and command line flag would be provided:
```

Comment:
Should state who would be providing the new configuration.

Reply:
This work requires a number of changes to the Cosmos SDK codebase, including the default configuration for client apps. New command line flags need to be made available for command line utilities development.